internal/builder/builder.go

```diff
@@ -520,28 +511,17 @@ func (b *Builder) defaultDirsLayer(dest string) (string, error) {
 
 	ts := archive.NormalizedDateTime
 
-	if err := lw.WriteHeader(b.packOwnedDir(workspaceDir, ts)); err != nil {
-		return "", errors.Wrapf(err, "creating %s dir in layer", style.Symbol(workspaceDir))
-	}
-
-	if err := lw.WriteHeader(b.packOwnedDir(layersDir, ts)); err != nil {
-		return "", errors.Wrapf(err, "creating %s dir in layer", style.Symbol(layersDir))
-	}
-
-	if err := lw.WriteHeader(b.rootOwnedDir(cnbDir, ts)); err != nil {
-		return "", errors.Wrapf(err, "creating %s dir in layer", style.Symbol(cnbDir))
-	}
-
-	if err := lw.WriteHeader(b.rootOwnedDir(dist.BuildpacksDir, ts)); err != nil {
-		return "", errors.Wrapf(err, "creating %s dir in layer", style.Symbol(dist.BuildpacksDir))
-	}
-
-	if err := lw.WriteHeader(b.rootOwnedDir(platformDir, ts)); err != nil {
-		return "", errors.Wrapf(err, "creating %s dir in layer", style.Symbol(platformDir))
+	for _, path := range []string{workspaceDir, layersDir} {
+		if err := lw.WriteHeader(b.packOwnedDir(path, ts)); err != nil {
+			return "", errors.Wrapf(err, "creating %s dir in layer", style.Symbol(path))
+		}
 	}
 
-	if err := lw.WriteHeader(b.rootOwnedDir(platformDir+"/env", ts)); err != nil {
-		return "", errors.Wrapf(err, "creating %s dir in layer", style.Symbol(platformDir+"/env"))
+	// can't use filepath.Join(), to ensure Windows doesn't transform it to Windows join
```

Comment:
FYI: `path.Join` works in place of `filepath.Join` when wanting to retain linux like functionality.

Reply:
That's good to know! I still think that having an explicit comment is helpful, otherwise someone may switch it from `path.Join` to `filepath.Join`. If you'd like me to change it, though, I'm happy to do that before merging it. LMK